bruce-pain/FastAPI-boilerplate
# build: add Poetry for dependency management and packaging

## Description
This PR introduces Poetry as the dependency management and packaging tool for the project. It replaces or complements any previous methods of managing dependencies, offering a modern, streamlined approach to handle project dependencies and virtual environments.

## Changes
Added Poetry as the primary dependency management tool.
Initialized the pyproject.toml file with project metadata and dependencies.
Documented usage instructions for Poetry in the README.md (if applicable).

## Why this change?
Improved Dependency Management: Poetry simplifies the process of managing dependencies with a declarative pyproject.toml file.
Enhanced Reproducibility: By generating a poetry.lock file, it ensures consistent environments across different systems.
Modern Python Packaging: Aligns the project with modern Python standards (PEP 518).

## How to Test
Install Poetry:
```bash
curl -sSL https://install.python-poetry.org | python3 -
```

Install dependencies:
```bash
poetry install
```

Run the project to ensure everything works as expected.
